spec/lib/groupers/drug_grouper_spec.rb

```diff
@@ -148,11 +179,50 @@ def create_chembl_drug_attributes
 
   end
 
-  it 'should add a drug if the drug claim matches a molecule alias, and add the drug claim to the drug' do
-    nil
-  end
-
   it 'should properly update drug flags after grouping' do
-    nil
+    approved_molecule = Fabricate(:chembl_molecule, max_phase: 4, withdrawn_flag: false)
+    withdrawn_molecule = Fabricate(:chembl_molecule, max_phase: 4, withdrawn_flag: true)
+    unapproved_molecule = Fabricate(:chembl_molecule, max_phase: 2, withdrawn_flag: false)
+
+    approved_drug = Fabricate(:drug, chembl_molecule: approved_molecule)
+    withdrawn_drug = Fabricate(:drug, chembl_molecule: withdrawn_molecule)
+    unapproved_drug = Fabricate(:drug, chembl_molecule: unapproved_molecule)
+
+    expect(approved_drug.fda_approved).to be_truthy
+    expect(withdrawn_drug.fda_approved).to be_falsey
+    expect(unapproved_drug.fda_approved).to be_falsey
+
+    immunotherapy_molecule = Fabricate(:chembl_molecule, molecule_type: 'Antibody')
+    small_molecule = Fabricate(:chembl_molecule, molecule_type: 'Small molecule')
+
+    immunotherapy_drug = Fabricate(:drug, chembl_molecule: immunotherapy_molecule)
+    non_immunotherapy_drug = Fabricate(:drug, chembl_molecule: small_molecule)
+
+    expect(immunotherapy_drug.immunotherapy).to be_truthy
+    expect(non_immunotherapy_drug.immunotherapy).to be_falsey
+
+    anti_neoplastic_drugs =
+      %w[TALC ClearityFoundationClinicalTrial ClearityFoundationBiomarkers CancerCommons
+                                      MyCancerGenome CIViC MyCancerGenomeClinicalTrial].map do |source_db_name|
+        source = Fabricate(:source, source_db_name: source_db_name)
+        drug_claim = Fabricate(:drug_claim, source: source)
+        drug = Fabricate(:drug, name: drug_claim.primary_name)
+    end
+
+    source = Fabricate(:source, source_db_name: 'ChEMBL')
+    chembl_drug_claim = Fabricate(:drug_claim, source: source)
+    chembl_drug = Fabricate(:drug, name: chembl_drug_claim.primary_name)
+
+    grouper = Genome::Groupers::DrugGrouper.new
+    grouper.run
+    anti_neoplastic_drugs.each do |drug|
+      drug.reload
+      expect(drug.drug_claims.count).to eq 1
+      drug.send(:update_anti_neoplastic_add, drug.drug_claims.first) # Why does this need to be done for rspec?
```

Comment:
There's something not quite right here. In a Rails console session, I can get the `Drug` `after_add` callback to work properly when adding a drug claim to a drug, but in rspec I need to call the associated private method manually to do the update.

```ruby
Loading development environment (Rails 5.1.1)
[1] pry(main)> >> drug = DataModel::Drug.new
=> #<DataModel::Drug:0x007f9264832658
 id: nil,
 name: nil,
 fda_approved: nil,
 immunotherapy: nil,
 anti_neoplastic: nil,
 chembl_id: nil>
[2] pry(main)> >> generic_source = DataModel::Source.where(source_db_name: 'ChEMBL').first
<Model::Source.where(source_db_name: 'ChEMBL').first
  DataModel::Source Load (0.9ms)  SELECT  "sources".* FROM "sources" WHERE "sources"."source_db_name" = $1 ORDER BY "sources"."id" ASC LIMIT $2  [["source_db_name", "ChEMBL"], ["LIMIT", 1]]
=> #<DataModel::Source:0x007f925d50fb80
 id: "30409f4e-3eb6-46fd-9661-9e8bfd6f525f",
 source_db_name: "ChEMBL",
 source_db_version: "chembl_20",
 citation:
  "The ChEMBL bioactivity database: an update. Bento AP, Gaulton A, Hersey A, Bellis LJ, Chambers J, Davies M, Krüger FA, Light Y, Mak L, McGlinchey S, Nowotka M, Papadatos G, Santos R, Overington JP. Nucleic Acids Res. 42(Database issue):D1083-90. PubMed ID: 24214965",
 base_url: "https://www.ebi.ac.uk/chembldb/index.php/target/inspect/",
 site_url: "https://www.ebi.ac.uk/chembl",
 full_name: "The ChEMBL Bioactivity Database",
 source_type_id: "ce28974a-b813-4717-a3d2-2df0cc54ab5a",
 gene_claims_count: 268,
 drug_claims_count: 906,
 interaction_claims_count: 1192,
 interaction_claims_in_groups_count: 0,
 gene_claims_in_groups_count: 252,
 drug_claims_in_groups_count: 748,
 source_trust_level_id: "600d121e-6260-4db8-9689-e72a13cbf2c3",
 gene_gene_interaction_claims_count: 0>
[3] pry(main)> >> cancer_source = DataModel::Source.where(source_db_name: 'TALC').first
<taModel::Source.where(source_db_name: 'TALC').first
  DataModel::Source Load (1.2ms)  SELECT  "sources".* FROM "sources" WHERE "sources"."source_db_name" = $1 ORDER BY "sources"."id" ASC LIMIT $2  [["source_db_name", "TALC"], ["LIMIT", 1]]
=> #<DataModel::Source:0x007f925f6b19c8
 id: "0009971b-b332-44a2-82a7-45128289099d",
 source_db_name: "TALC",
 source_db_version: "12-May-2016",
 citation:
  "A tabulated summary of targeted and biologic therapies for non-small-cell lung cancer. Simon GR, Somaiah N. Clin Lung Cancer. 2014 Jan;15(1):21-51. doi: 10.1016/j.cllc.2013.11.009. Epub 2013 Nov 21. PMID: 24377743",
 base_url: "http://www.ncbi.nlm.nih.gov/gene?term=",
 site_url: "https://www.ncbi.nlm.nih.gov/pubmed/24377743",
 full_name: "Targeted Agents in Lung Cancer (Commentary, 2014)",
 source_type_id: "ce28974a-b813-4717-a3d2-2df0cc54ab5a",
 gene_claims_count: 0,
 drug_claims_count: 0,
 interaction_claims_count: 0,
 interaction_claims_in_groups_count: 0,
 gene_claims_in_groups_count: 0,
 drug_claims_in_groups_count: 0,
 source_trust_level_id: "600d121e-6260-4db8-9689-e72a13cbf2c3",
 gene_gene_interaction_claims_count: 0>
[4] pry(main)> >> dc = DataModel::DrugClaim.new(source: generic_source)
=> #<DataModel::DrugClaim:0x007f925fde9588
 id: nil,
 name: nil,
 nomenclature: nil,
 source_id: "30409f4e-3eb6-46fd-9661-9e8bfd6f525f",
 primary_name: nil,
 drug_id: nil>
[5] pry(main)> >> cancer_dc = DataModel::DrugClaim.new(source: cancer_source)
=> #<DataModel::DrugClaim:0x007f925fdc65d8
 id: nil,
 name: nil,
 nomenclature: nil,
 source_id: "0009971b-b332-44a2-82a7-45128289099d",
 primary_name: nil,
 drug_id: nil>
[6] pry(main)> >> drug.drug_claims << dc
=> [#<DataModel::DrugClaim:0x007f925fde9588
  id: nil,
  name: nil,
  nomenclature: nil,
  source_id: "30409f4e-3eb6-46fd-9661-9e8bfd6f525f",
  primary_name: nil,
  drug_id: nil>]
[8] pry(main)> >> drug.anti_neoplastic
=> false
[9] pry(main)> >> drug.drug_claims << cancer_dc
=> [#<DataModel::DrugClaim:0x007f925fde9588
  id: nil,
  name: nil,
  nomenclature: nil,
  source_id: "30409f4e-3eb6-46fd-9661-9e8bfd6f525f",
  primary_name: nil,
  drug_id: nil>,
 #<DataModel::DrugClaim:0x007f925fdc65d8
  id: nil,
  name: nil,
  nomenclature: nil,
  source_id: "0009971b-b332-44a2-82a7-45128289099d",
  primary_name: nil,
  drug_id: nil>]
[10] pry(main)> >> drug.anti_neoplastic
=> true
[11] pry(main)> >> drug.drug_claims.last.drug_id = nil
=> nil
[12] pry(main)> >> drug.anti_neoplastic
=> true
[13] pry(main)> >> drug.drug_claims
=> [#<DataModel::DrugClaim:0x007f925fde9588
  id: nil,
  name: nil,
  nomenclature: nil,
  source_id: "30409f4e-3eb6-46fd-9661-9e8bfd6f525f",
  primary_name: nil,
  drug_id: nil>,
 #<DataModel::DrugClaim:0x007f925fdc65d8
  id: nil,
  name: nil,
  nomenclature: nil,
  source_id: "0009971b-b332-44a2-82a7-45128289099d",
  primary_name: nil,
  drug_id: nil>]
```
Also, I noticed that for _removing_ drug claims from a drug, the `before_remove` callback also doesn't seem to work, even from console, unless the private method is called directly from the object:
```ruby
[14] pry(main)> >> drug.drug_claims.delete(cancer_dc)
=> [#<DataModel::DrugClaim:0x007f925fdc65d8
  id: nil,
  name: nil,
  nomenclature: nil,
  source_id: "0009971b-b332-44a2-82a7-45128289099d",
  primary_name: nil,
  drug_id: nil>]
[15] pry(main)> >> drug.anti_neoplastic
=> true
[17] pry(main)> >> drug.send(:update_anti_neoplastic_remove, 'foobar')
=> false
[18] pry(main)> >> drug.anti_neoplastic
=> false
```

@acoffman, could you shed some light on what's going on here? I don't think that this is a show-stopper for our current work, but it would be nice for my understanding as to why these specific callbacks seem to be finicky.